.github/workflows/docker-publish.yml

```diff
@@ -0,0 +1,43 @@
+name: Publish on Github container registry
+
+on:
+  release:
+    type: [published]
```

Comment:
Yes it makes sense to push a new `gimie:latest` image for every commit on master! However, I think we should also push a new minor release (e.g., from `vX.Y.1` to `vX.Y.2`) at the same time for the following reason:

If we only push new `latest` images, we end up with `latest` images that do not match any other version. This is a problem because using the `latest` tag for production is a very bad practice, which makes the last gimie updates (i.e., what's on the `main` branch) unusable for production environments. 

I propose the following:
- If the changes are impacting the behavior (e.g., bug fixes, new features or syntactic changes), we should push a new minor release
- If the changes are only visual (e.g., fixing typos in texts/comments/logs) we should push to the same `vX.Y.Z` version as before. If we do not want to fully replace the previous images, we can additionally push images with the date in the tags as [Ubuntu](https://hub.docker.com/_/ubuntu) does, for example. But that makes yet another image to push and I don't know if we are limited in terms of size on the registry

What do you think?

Reply:
Good point! Bumping the minor version for _every_ commit on master sounds a bit tedious.
Appending build information to the image version sounds good. A [semver](https://semver.org/)-compatible way of achieving this is to use commit hashes, indeed:
`<version core> "+" <build>`
Where <version core> is the last official release, and <build> is the current commit hash. (e.g. `v0.2.0+bb9dcb8`).